hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/IndexCommands.scala

```diff
@@ -51,6 +53,15 @@ case class CreateIndexCommand(table: CatalogTable,
       new util.LinkedHashMap[String, java.util.Map[String, String]]()
     columns.map(c => columnsMap.put(c._1.mkString("."), c._2.asJava))
 
+    // validate that only overwrite with latest payloads can enabled SI
+    if (indexType.equals("secondary_index")) {
+      // do additional validation
+      if ((metaClient.getTableConfig.getPayloadClass != null && !(metaClient.getTableConfig.getPayloadClass.equals(classOf[OverwriteWithLatestAvroPayload].getCanonicalName)))
+        || (metaClient.getTableConfig.getRecordMergeMode ne RecordMergeMode.OVERWRITE_WITH_LATEST)) {
```

Comment:
Should we also consider `DefaultHoodieRecordPayload` if event time field is not specified?

Reply:
Discussed offline. Not needed. We can land this once the CI passes.